### ISSUE TITLE
When `ChatEventHandler.onMessageNew()` get invoked with new message - `currentUserUnreadMessageCount` returns `0` for some channels.

### ISSUE THREAD

**Issue by AlexWih** [GetStream/stream-chat-android#469]
I have following (simplified) code: 
```
val client = StreamChat.getInstance(applicationContext)
//...  querying all channels using pagination described here: https://github.com/GetStream/stream-chat-android/issues/462#issuecomment-627218496

client.addEventHandler(object : ChatEventHandler() {
    override fun onMessageNew(channel: Channel, event: Event) {
        super.onMessageNew(channel, event)
        logger.log("onMessageNew(), channel ID: ${channel.cid}")
        logger.log("onMessageNew(), unread count: ${channel.lastState?.currentUserUnreadMessageCount}, count2: ${channel.channelState?.currentUserUnreadMessageCount}")
    }

    override fun onNotificationMessageNew(channel: Channel, event: Event) {
        super.onNotificationMessageNew(channel, event)
        logger.log("onNotificationMessageNew(), channel ID: ${channel.cid}")
    }

    override fun onMessageRead(channel: Channel, event: Event) {
        super.onMessageRead(channel, event)
        logger.log("onMessageRead(), unread count: ${channel.lastState?.currentUserUnreadMessageCount}")
    }
})
```

I send the message from another device and cb `onMessageNew()` gets invoked, the following logs are printed:
```
2020-05-12 12:26:49.487 9030-9030/com.trember.myapplication D/StreamSocketWrapperImpl: onMessageNew(), channel ID: messaging:d899a7f0-b211-4875-8e86-5d23db76f85b
2020-05-12 12:26:49.487 9030-9030/com.trember.myapplication D/StreamSocketWrapperImpl: onMessageNew(), unread count: 0, count2: 0
```

Then I send message from the same device to the different channel and cb `onMessageNew()` is also invoked: 
```
2020-05-12 12:32:37.985 9030-9030/com.trember.myapplication D/StreamSocketWrapperImpl: onMessageNew(), channel ID: messaging:9ea6cf03-6c6a-4372-82b6-ee37db5927e4
2020-05-12 12:32:37.985 9030-9030/com.trember.myapplication D/StreamSocketWrapperImpl: onMessageNew(), unread count: 3, count2: 3
```

As you can see in the 1st case neither `channel.lastState.currentUserUnreadMessageCount()` nor 
`channel.channelState?.currentUserUnreadMessageCount()` provides real value - it is always 0, but for other channels I do receive unread count.

I am expecting that if cb `onMessageNew()` is invoked - the `currentUserUnreadMessageCount()` should return its incremented previous value as we just received the new message which user hasn't had a chance to read yet, but it seems it doesn't work for some channels.

**Comment by tschellenbach** [comment]
I asked Eugene to look into this. In general I recommend the beta/4.x branch if possible.

**Comment by samiuelson** [comment]
Hi @AlexWih the issue is already fixed in the latest SDK release [4.40](https://github.com/GetStream/stream-chat-android/releases/tag/4.4.0). 
Feel free to reopen if you have any further problems.
